include/zephyr/sys/util.h

```diff
@@ -366,9 +366,10 @@ extern "C" {
  *
  * @return The result of @p n / @p d, rounded to the nearest integer.
  */
-#define DIV_ROUND_CLOSEST(n, d)	\
-	((((n) < 0) ^ ((d) < 0)) ? ((n) - ((d) / 2)) / (d) : \
-	((n) + ((d) / 2)) / (d))
+#define DIV_ROUND_CLOSEST(n, d)                                                                    \
+	((!(((__typeof__(n))-1) > 0) && !(((__typeof__(d))-1) > 0) && ((n) < 0) ^ ((d) < 0))       \
```

Comment:
(!(((__typeof__(n))-1) > 0) && !(((__typeof__(d))-1) > 0)

Reply:
```suggestion
	(((((__typeof__(n))-1) < 0) && (((__typeof__(d))-1) < 0) && ((n) < 0) ^ ((d) < 0))       \
```

Reply:
> (!(((**typeof**(n))-1) > 0) && !(((**typeof**(d))-1) > 0)

Only __typeof__ is supported by the toolchain